src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs

```diff
@@ -156,7 +156,7 @@ public Task ConnectAsync(Uri uri, CancellationToken cancellationToken)
 
         private Task ConnectAsyncCore(Uri uri, CancellationToken cancellationToken)
         {
-            _innerWebSocket = new WinHttpWebSocket();
+            _innerWebSocket = WebSocketHandle.Create();
 
```

Comment:
Ah, that's clearer. :) I'm happy to wait for the code to flow in from TFS and adjust this change accordingly as long as @roncain is OK with waiting. @roncain: in the meantime, you could pull down https://github.com/pgavlin/corefx/tree/WebSocketClientPNSE and use the S.N.WebSockets.Client.dll out of the build.

(/cc @joshfree just as a heads-up)